jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInitRunner.kt

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.CapturingProcessHandler
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.settings.SamSettings
+import software.aws.toolkits.resources.message
+
+class SamInitRunner(
+    private val name: String,
+    private val outputDir: VirtualFile,
+    private val runtime: Runtime,
+    private val location: String? = null
+) {
+    private val samCliExecutable = SamSettings.getInstance().executablePath
+    private val commandLine = GeneralCommandLine(samCliExecutable)
+            .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.CONSOLE)
+            .withParameters("init")
+            .withParameters("--no-input")
+            .withParameters("--name")
+            .withParameters(name)
+            .withParameters("--runtime")
+            .withParameters(runtime.toString())
+
+    fun execute() = ApplicationManager.getApplication().runWriteAction {
+        // set output to a temp dir
+        val tempDir = LocalFileSystem.getInstance().findFileByIoFile(createTempDir())
+                ?: throw RuntimeException("Cannot create temp file")
+        val commandLine = commandLine.withParameters("--output-dir")
```

Comment:
Nit: it's a bit confusing to have the same name for the two variables. Consider rename it or just move the `commandLine` in the `execute` method as there are no other methods in the class. There are no much point for having this member.